codegen/smithy-aws-ruby-codegen/src/main/java/software/amazon/smithy/aws/ruby/codegen/AWSIntegration.java

```diff
@@ -133,14 +87,18 @@ metadata service (IMDS) on an EC2 instance.
                 .shapeId(SigV4Trait.ID)
                 .rubyAuthScheme("AWS::SDK::Core::AuthSchemes::SigV4.new")
                 .rubyIdentityType(awsCredentialsIdentity.toString())
-                .identityProviderConfig(identityProviderConfig)
+                .identityProviderConfig(AWSConfig.CREDENTIALS_PROVIDER)
                 .extractSignerProperties((trait -> {
                     SigV4Trait sigv4 = (SigV4Trait) trait;
                     return Map.of(
                             "service", "'" + sigv4.getName() + "'",
                             "region", "params.region"
                     );
                 }))
+                .additionalConfig(AWSConfig.ACCESS_KEY_ID)
```

Comment:
I'm not sure how I feel about including these.  Users can just configure a static credential provider to get these.  But I guess we do have them in V3 so having them here does provide better backwards compatability... (Though we could also provide that through a compat gem/shim that adds a config plugin or something that would convert them or some such).

Reply:
Check out the quip on the python doc. It says these are a must and I think it makes sense because it powers the from code provider.